### PR TITLE
Fix mutini sometimes removing abilities you already have or your mutantrace

### PR DESCRIPTION
**Pull request by thegrb93**
## About the PR 
Fix mutini sometimes removing abilities you already have or your mutantrace

## Why's this needed? 
Mutini is very annoying if it ends up randomly removing your mutations.

## Changelog 

```changelog
(u) Lizzy
(+) Fix mutini sometimes removing abilities you already have or your mutantrace
```

### CODE REVIEW

**Review by Tarmunora — CHANGES_REQUESTED** [review]
would rather just cache the original mutantrace and restore it in on_mob_life_complete or whatever

**Comment by thegrb93** [comment]
Would it be a var stored on the mob? M.mutini_original_race?

**Comment by thegrb93** [comment]
@Tarmunora one issue I foresee with that is you'll turn human until the mutini 'complete' is called and then turn back to your mutantrace.

**Comment by thegrb93** [comment]
Maybe if the bioeffect is of the mutantrace type, I can make it apply and unapply with the original race. If two or more races get triggered at the same time though, I'd have to make sure it doesn't get messy.

**Comment by thegrb93** [comment]
Finished. Will test it in a few hours.

**Comment by thegrb93** [comment]
Got it working and tested with lots of mutini

File: code/mob/living/carbon/human.dm [hunk 1]
```diff
@@ -87,6 +87,8 @@
 	var/uses_damage_overlays = 1 //If set to 0, the mob won't receive any damage overlays.
 
 	var/datum/mutantrace/mutantrace = null
+	/// used by mutini to store and restore what you are while randomizing mutations
+	var/datum/mutantrace/mutini_orig_mutantrace = null
```
Comment (Tarmunora) [inline code/mob/living/carbon/human.dm:91]: should be on the reagent, not the mob

File: code/modules/chemistry/Reagents-FoodDrink.dm [hunk 2]
```diff
@@ -861,32 +861,41 @@ datum
...
+					return
 				var/power_time = rand(1,10)
-				M.bioHolder.AddEffect(power_granted)//, 0, power_time)
+				var/is_mutant_race = istype(GetBioeffectFromGlobalListByID(power_granted), /datum/bioEffect/mutantrace)
+				if(is_mutant_race)
+					if(M.mutini_orig_mutantrace)
+						return // We're already some mutini mutantrace
+					if(M.mutantrace)
+						M.mutini_orig_mutantrace = M.mutantrace
+
```
Comment (Tarmunora) [inline code/modules/chemistry/Reagents-FoodDrink.dm:891]: set this in on_add instead

File: code/modules/chemistry/Reagents-FoodDrink.dm [hunk 3]
```diff
@@ -861,32 +861,41 @@ datum
...
+					
+				M.bioHolder.AddEffect(power_granted)//, 0, power_time) the timeLeft var either wasn't working here or was grumpy about something so now we manually remove this below
 				SPAWN(power_time*10)
 					if (M?.bioHolder)
 						M.bioHolder.RemoveEffect(power_granted)
-				..()
-				return
+						if(is_mutant_race && M.mutini_orig_mutantrace)
+							M.set_mutantrace(M.mutini_orig_mutantrace)
+							M.mutini_orig_mutantrace = null
```
Comment (Tarmunora) [inline code/modules/chemistry/Reagents-FoodDrink.dm:898]: handle in on_mob_life_complete()

**Review by Tarmunora — CHANGES_REQUESTED** [review]
> @Tarmunora one issue I foresee with that is you'll turn human until the mutini 'complete' is called and then turn back to your mutantrace.

mutini fucks up your genes a lot while it's in you. It's fine for it to not preserve mutantrace while it's still doing it's thing.

File: code/mob/living/carbon/human.dm (hunk 1 above)
Comment (Tarmunora) [reply code/mob/living/carbon/human.dm:91]: no

**Comment by thegrb93** [comment]
Tried it out, giving every monkey and myself mutini. Looks like it's working.

File: code/modules/chemistry/Reagents-FoodDrink.dm [hunk 4]
```diff
@@ -859,35 +859,51 @@ datum
...
+					return
+				var/power_granted = pick(mutini_effects)
+				if (M.bioHolder.HasEffect(power_granted))
+					return
+				var/power_time = rand(1,10)
+				var/is_mutant_race = istype(GetBioeffectFromGlobalListByID(power_granted), /datum/bioEffect/mutantrace)
+
+				M.bioHolder.AddEffect(power_granted)//, 0, power_time) the timeLeft var either wasn't working here or was grumpy about something so now we manually remove this below
+				// Keep whatever mutant races we get until mutini wears off
+				if(!is_mutant_race)
```
Comment (thegrb93) [inline code/modules/chemistry/Reagents-FoodDrink.dm:902]: Probably need to remove the mutant race bioeffect still but without removing the mutantrace.

Comment (Tarmunora) [reply code/modules/chemistry/Reagents-FoodDrink.dm:902]: mutantrace bioeffects remove themself if the owner changes mutantrace

**Review by Tarmunora — CHANGES_REQUESTED** [review]

**Comment by thegrb93** [comment]
Will test again later tonight

**Comment by thegrb93** [comment]
Works

**Review by Tarmunora — APPROVED** [review]

**Pull request merged**